src/manifolds/MetricManifold.jl

```diff
@@ -1,8 +1,15 @@
 @doc raw"""
     Metric
 
-Abstract type for the pseudo-Riemannian metric tensor $g$, a family of smoothly
+Abstract type for the pseudo-Riemannian metric tensor ``g``, a family of smoothly
 varying inner products on the tangent space. See [`inner`](@ref).
+
+# Constructors
+
+    Metric(M)
+
+Generate the manifold `M` with the correstponding metric (subtype of this type).
```

Comment:
```suggestion
Generate the manifold `M` with the corresponding metric (subtype of this type).
```
Do you mean `MetricManifold` by "this type" or something else?

Reply:
Good that you ask, no, I meant the `Metric`. I wanted to clarify that there is `Metric(M)` for `MetricManifold(M,Metric())` as a constructor.

Reply:
Then the documentation is quite unclear, and this isn't exactly a constructor of `Metric` type since it returns a value of type `MetricManifold`. It should be more clearly stated I think. First, I'd add that it required a concrete subtype of `Metric` in the call, a second the description could be a bit more like:
```
    (metric::Metric)(M::Manifold)

Generate the `MetricManifold` that wraps the manifold `M` with given `metric`.
```

Reply:
Or similarly for the other method, `(::Type{T})(M::AbstractManifold) where {T<:AbstractMetric}`.

Reply:
Ah yes, that is a better signature. By the way, just for completeness we could also introduce `(M::Manifold)(m::Metric)` to do the same (which even feels more natural to me).

Reply:
> By the way, just for completeness we could also introduce `(M::Manifold)(m::Metric)` to do the same (which even feels more natural to me).

I don't have any opinions on that but if you like it then sure :slightly_smiling_face: .

Reply:
Maybe in a later PR to not make this PR too chaotic.